edx_rest_framework_extensions/auth/jwt/authentication.py

```diff
@@ -106,6 +113,20 @@ def authenticate(self, request):
                 set_custom_attribute('jwt_auth_result', 'n/a')
                 return user_and_auth
 
+            lms_user_id_property_name = get_setting(VERIFY_LMS_USER_ID_PROPERTY_NAME)
+            # Check lms_user_id_property_name to be 'id' to make this check work for LMS only
+            if lms_user_id_property_name == 'id':
+                user, auth = user_and_auth
+                is_email_mismatch = self._is_jwt_and_lms_user_email_mismatch(request, user)
+                if is_email_mismatch and get_setting(ENABLE_JWT_AND_LMS_USER_EMAIL_MATCH):
+                    set_custom_attribute('jwt_auth_result', 'user-email-mismatch-failure')
+                    raise JwtUserEmailMismatchError(
+                        'Failing JWT authentication due to jwt user email mismatch '
+                        'with lms user email.'
+                    )
+                elif is_email_mismatch:
+                    set_custom_attribute('jwt_auth_result', 'user-email-mismatch-failure')
```

Comment:
Two changes:
1. I added a new and separate custom attribute `jwt_email_mismatch`.
2. I removed the `elif`, because `jwt_auth_result` will be overridden lower down based on the actual result.
3. Nit: To keep this method cleaner, I'd move the `VERIFY_LMS_USER_ID_PROPERTY_NAME` check inside `_is_jwt_and_lms_user_email_mismatch`, and just return early. 
```suggestion
            is_email_mismatch = self._is_jwt_and_lms_user_email_mismatch(request, user_and_auth[0])
            if is_email_mismatch:
                # .. custom_attribute_name: jwt_email_mismatch
                # .. custom_attribute_description: Temporary attribute to show if email mismatched
                #     even when the enforce toggle is disabled.
                set_custom_attribute('jwt_email_mismatch', is_email_mismatch)
                if get_setting(ENABLE_JWT_AND_LMS_USER_EMAIL_MATCH):
                    set_custom_attribute('jwt_auth_result', 'user-email-mismatch-failure')
                    raise JwtUserEmailMismatchError(
                        'Failing JWT authentication due to jwt user email mismatch '
                        'with lms user email.'
                    )
```

Reply:
@syedsajjadkazmii: Ignore the code change I made above. I've changed my mind on some things:
1. Do not use `VERIFY_LMS_USER_ID_PROPERTY_NAME`. It may go away.
2. `ENABLE_JWT_AND_LMS_USER_EMAIL_MATCH` will need to be a permanent toggle, and should be documented as such. Although it will be a temporary rollout toggle for the LMS, ultimately it will be a permanent toggle for the LMS (as the identity server), and will be disabled for all other services.
3. To simplify the code, since this will be permanently disabled elsewhere, you can drop the `jwt_email_mismatch` custom attribute. This means you won't be able to get observability in the LMS before enabling, but hopefully previous testing will be enough.

So code might just look like:
```
if get_setting(ENABLE_JWT_AND_LMS_USER_EMAIL_MATCH):
    is_email_mismatch = self._is_jwt_and_lms_user_email_mismatch(request, user_and_auth[0])
    if is_email_mismatch:
        ...
```

Reply:
Hi @robrap,

Thanks for the suggestions. I have updated the PR with the latest comments. Please have a look.